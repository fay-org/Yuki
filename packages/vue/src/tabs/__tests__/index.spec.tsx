import { mount } from '@vue/test-utils'
import { later } from '../../../tests'
import Tabs from '../index'
import Tab from '../../tab'

describe('Tabs', () => {
  it('should be render as element', () => {
    const wrapper = mount(Tabs)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be change acitve value when tab trigger', async () => {
    const wrapper = mount({
      render() {
        return (
          <Tabs v-model={[this.active, ['active']]}>
            <Tab title="HTML">HTML</Tab>
            <Tab title="JS">JS</Tab>
          </Tabs>
        )
      },
      data() {
        return {
          active: 0
        }
      }
    })
    await later()
    const els = wrapper.findAll('.fect-tabs__title') // filter all tabs title
    await els[1].trigger('mouseenter')
    expect(wrapper.find('.fect-tabs__highlight--active')).toBeTruthy()
    const headerEl = wrapper.find('header')
    await headerEl.trigger('mouseleave')
    await els[1].trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be disabled all events', async () => {
    const wrapper = mount({
      render() {
        return (
          <Tabs v-model={[this.active, ['active']]} hideDivider={false}>
            <Tab title="HTML" disabled={this.disabled}>
              HTML
            </Tab>
            <Tab title="JS">JS</Tab>
          </Tabs>
        )
      },
      data() {
        return {
          active: 0,
          disabled: true
        }
      }
    })
    await later()
    const els = wrapper.findAll('.fect-tabs__title') // filter all tabs title
    await els[0].trigger('click')
    await wrapper.setProps({
      hideDivider: true
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Tab can not render without Tabs', () => {
    const wrapper = mount(Tab)
    expect(() => wrapper.get('div')).toThrowError()
  })
  it('Tabs can use slots label custom tab title', () => {
    const wrapper = mount(Tabs, {
      slots: {
        default: () => {
          return (
            <>
              <Tab v-slots={{ label: () => <div>Custom</div> }}>HTML</Tab>
              <Tab>JS</Tab>
            </>
          )
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
