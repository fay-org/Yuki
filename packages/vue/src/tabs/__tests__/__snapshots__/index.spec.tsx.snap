// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Tabs can use slots label custom tab title 1`] = `
<div class="fect-tabs">
  <header class="fect-tabs__header" role="tablist" aria-orientation="horizontal">
    <div class="fect-tabs__highlight" style="transition-property: opacity;"></div>
  </header>
  <div class="fect-tab" role="tabpanel" tabindex="0">HTML</div>
  <div class="fect-tab" role="tabpanel" tabindex="-1" style="display: none;">JS</div>
</div>
`;

exports[`Tabs should be change acitve value when tab trigger 1`] = `
<div class="fect-tabs" modelvalue="0" modelmodifiers="[object Object]">
  <header class="fect-tabs__header" role="tablist" aria-orientation="horizontal">
    <div class="fect-tabs__highlight" style="transition-property: opacity, width, left, top; opacity: 0; width: 0px; left: 0px; height: 0px; top: 0px;"></div>
    <div role="tab" class="fect-tabs__title">HTML</div>
    <div role="tab" class="fect-tabs__title fect-tabs__title--active">JS</div>
  </header>
  <div class="fect-tab" role="tabpanel" tabindex="-1" style="display: none;">HTML</div>
  <div class="fect-tab" role="tabpanel" tabindex="0" style="">JS</div>
</div>
`;

exports[`Tabs should be disabled all events 1`] = `
<div class="fect-tabs" modelvalue="0" modelmodifiers="[object Object]">
  <header class="fect-tabs__header fect-tabs__header--hide-divider" role="tablist" aria-orientation="horizontal">
    <div class="fect-tabs__highlight" style="transition-property: opacity;"></div>
    <div role="tab" class="fect-tabs__title fect-tabs__title--disabled fect-tabs__title--active">HTML</div>
    <div role="tab" class="fect-tabs__title">JS</div>
  </header>
  <div class="fect-tab" role="tabpanel" tabindex="0">HTML</div>
  <div class="fect-tab" role="tabpanel" tabindex="-1" style="display: none;">JS</div>
</div>
`;
